transform-plugins/src/main/java/io/cdap/plugin/Decompressor.java

```diff
@@ -68,6 +73,32 @@ public Decompressor(Config config) {
     this.config = config;
   }
 
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+    parseConfiguration(config.decompressor, context.getFailureCollector());
+
+    // Initialize the required member maps and then have a one-to-one transform on all fields present in
+    // outSchemaMap and deCompMap and aren't set to NONE as decompressorType.
+    List<String> inFields = TransformLineageRecorderUtils.getFields(context.getInputSchema());
+    List<String> outFields = TransformLineageRecorderUtils.getFields(context.getOutputSchema());
+    List<String> identityFields = outFields.stream()
+      .filter(field -> !deCompMap.containsKey(field) || deCompMap.get(field) == DecompressorType.NONE)
+      .collect(Collectors.toList());
+
+    List<String> processedFields = new ArrayList<>(outFields);
+    processedFields.removeAll(identityFields);
+    List<String> droppedFields = new ArrayList<>(inFields);
+    droppedFields.removeAll(outFields);
+
+    List<FieldOperation> output = TransformLineageRecorderUtils.generateOneToOnes(processedFields, "decompress",
```

Comment:
Initialize the output first, `List<FieldOperation> output = new ArrayList<>()`, since it is possible that the util method will return an immutable list. This applies to other places in this pr